options.go

```diff
@@ -152,3 +164,16 @@ func (o *PushOptions) Validate() error {
 
 	return nil
 }
+
+// SubmoduleUpdateOptions describes how a submodule update should be performed
+type SubmoduleUpdateOptions struct {
+	// Init initializes the submodules recorded in the index
```

Comment:
This needs much better explanation. It is a bool, but does not make sense with the comment.